### PR TITLE
Here is my code for linear Svm

### PR DESCRIPTION
**Pull request by hokie45**

### CODE REVIEW

File: docs/source/content/overview/linear_SVM.rst [hunk 1]
```diff
@@ -2,20 +2,24 @@
...
+----------
+A **hyperplane** depends on the space it is in, but it divides the space into two disconnected parts. So 1-dimensional space this would just be a point, 2-d space a line, 3-d space a plane, and so on. 
 
 How do we find the best hyperplane/line?
 ----------------------------------------
 
-You might be wondering that there could be multiple lines that split the data well. As you can see in the graph below every line splits the squares and the circles, so which one do we choose?
+You might be wondering that there could be multiple lines that split the data well. In fact there is a infinite amount of lines that can divide two classes.  As you can see in the graph below every line splits the squares and the circles, so which one do we choose?
 
 .. figure:: https://github.com/machinelearningmindset/machine-learning-for-everybody/blob/master/docs/source/content/overview/_img/Possible_hyperplane.PNG
```
Comment (BroccoliHijinx) [inline docs/source/content/overview/linear_SVM.rst]:
```suggestion
.. figure:: /_img/Possible_hyperplane.PNG
```

File: docs/source/content/overview/linear_SVM.rst [hunk 2]
```diff
@@ -54,13 +58,29 @@ There will be data classes that can't be separated with a simple line or hyperpl
    Ref:  https://www.analyticsvidhya.com/blog/2017/09/understaing-support-vector-machine-example-code/
 
 
-There's is no clear way to separate the stars from the circles. We will go over how SVMs will be able to classify non-lenearly separable data in our section about **Kernal SVM**. 
+There's is no clear way to separate the stars from the circles. SVMs will be able to classify non-lenearly separable data by using a trick called the **kernel trick**. Basically what goes behind the scenes, is that it takes the points to a higher dimension, which turns non-linearly separable date to linear seprarable data. So the the above figure would be classified with a circle that separates the data.
```
Comment (BroccoliHijinx) [inline docs/source/content/overview/linear_SVM.rst]:
```suggestion
There's is no clear way to separate the stars from the circles. SVMs will be able to classify non-linearly separable data by using a trick called the **kernel trick**. Basically what goes behind the scenes, is that it takes the points to a higher dimension, which turns non-linearly separable date to linear separable data. So the the above figure would be classified with a circle that separates the data. 
```

Comment (BroccoliHijinx) [reply docs/source/content/overview/linear_SVM.rst]:
```suggestion
There's is no clear way to separate the stars from the circles. SVMs will be able to classify non-linearly separable data by using a trick called the **kernel trick**. Basically what goes behind the scenes, is that it takes the points to a higher dimension, which turns non-linearly separable date to linear separable data. So the above figure would be classified with a circle that separates the data. 
```

File: docs/source/content/overview/linear_SVM.rst [hunk 3]
```diff
@@ -2,20 +2,24 @@
...
+A **Support Vector Machine** (SVM for short) is another machine learning algorithm that is used to classify data. The point of SVM's are to try and find a line or **hyperplane** to divide a dimensional space which best classifies the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B. This alogorithm is very useful in classifying because we just have to caluclate the best line or hyperplane once and any knew data points we can easily classify just by seeing which side of the line it falls on. Unlike KNN, where we would have to calculate each data points nearest neighbors.
```
Comment (BroccoliHijinx) [inline docs/source/content/overview/linear_SVM.rst]:
```suggestion
A **Support Vector Machine** (SVM for short) is another machine learning algorithm that is used to classify data. The point of SVM's are to try and find a line or **hyperplane** to divide a dimensional space which best classifies the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B. This alogorithm is very useful in classifying because we just have to caluclate the best line or hyperplane once and any new data points we can easily classify just by seeing which side of the line it falls on. Unlike KNN, where we would have to calculate each data points nearest neighbors. 
```

**Review by astorfi — CHANGES_REQUESTED** [review]
Thank you guys for your work ... @weric16 Please clearly comment on documentation and the code ... @hokie45 Please change the image sizes in the documentation. Organize the document and link to your code within your documents. You can explain your code in your document so please can follow the concept. Make sure the code will generate the results promptly and add the output of your code in your documentation. When you create a pull request, please address how to took care of the points that I mentioned.

**Review by astorfi — APPROVED** [review]

**Review by astorfi — CHANGES_REQUESTED** [review]
Thank you for correcting the image sizes. Please take care of the motivation section and the code itself.

File: code/linear_svm.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+#All the libraries we need for linear SVM
```
Comment (mergu) [inline code/linear_svm.py]: All python comments should have a space after the # for cleanliness & consistency

File: code/linear_svm.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+# =============================================================================
+# We are using sklearn datasets to create the set of data points about breast cancer
+# Data is the set data points
+# target is the classification of those data points. 
+# More information can be found athttps://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_breast_cancer.html#sklearn.datasets.load_breast_cancer
+# =============================================================================
+dataCancer = load_breast_cancer()
+data = dataCancer.data[:, :2]
```
Comment (mergu) [inline code/linear_svm.py]: What does the `[:, :2]` do? Probably a good idea to explain that or rewrite it in a clearer way

File: code/linear_svm.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+# Plot the decision line and the margins
+axis.contour(X, Y,  decision_line, colors = 'k',  levels=[0], 
+           linestyles=['-'])
+# Shows the support vectors that determine the desision line
+axis.scatter(model.support_vectors_[:, 0], model.support_vectors_[:, 1], s=100,
+           linewidth=1, facecolors='none', edgecolors='k')
+
+#Shows the graph
+plt.show()
```
Comment (mergu) [inline code/linear_svm.py]: Need a newline at end of file here

File: docs/source/content/overview/linear_SVM.rst [hunk 7]
```diff
@@ -2,67 +2,123 @@
...
+A **Support Vector Machine** (SVM for short) is another machine learning algorithm that is used to classify data. The point of SVM's are to try and find a line or **hyperplane** to divide a dimensional space which best classifies the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B. This alogorithm is very useful in classifying because we just have to caluclate the best line or hyperplane once and any new data points we can easily classify just by seeing which side of the line it falls on. Unlike KNN, where we would have to calculate each data points nearest neighbors.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: All of your documentation should have line length limits. Typically this is 80-100 chars. Otherwise, this is just one very long string which becomes annoying to work with in some editors. I also can't comment on short sections of sentences with it as one line

File: docs/source/content/overview/linear_SVM.rst [hunk 8]
```diff
@@ -2,67 +2,123 @@
...
+You can see how these kernels change the outcome of the optimal hyperplane by changing the value of kernel in "model = svm.SVC(kernel = 'linear', C = 10000)" to either 'poly' or 'rbf'. This is in the linear_svm.py. 
 
 
 Conclusion
 -----------
 
-A SVM is a great machine learning technique to classify data. The pros to SVM's are it is a great classifier when there are mulitple dimensions in the data points. It is also saves space on memory because it only uses the support vectors to create the optimal line. The cons to SVM's are that the training times are higher when there is a larger data set, so it doesn't perform well. Also it performs badly when the classes are overlapping. 
+A SVM is a great machine learning technique to classify data. Now that we know a little about SVM's we can show the advantages and disadvantages to using this classifier. 
+The pros to SVM's:
+- Effective in classifying higher dimensional space
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst:112]: These lists do not appear correctly on the page because you need a whitespace between them and the previous paragraph

**Review by mergu — CHANGES_REQUESTED** [review]

File: docs/source/content/overview/linear_SVM.rst [hunk 9]
```diff
@@ -2,67 +2,157 @@
 Linear Support Vector Machines
 ==========================
 
-A **Support Vector Machine** (SVM for short) is another machine learning algorithm that is used to classify data. The point of SVM's are to try and find a line or hyperplane to divide a dimensional space which best classifies the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B. This alogorithm is very useful in classifying because we just have to caluclate the best line or hyperplane once and any knew data points we can easily classify just by seeing which side of the line it falls on. Unlike KNN, where we would have to calculate each data points nearest neighbors. 
+A **Support Vector Machine** (SVM for short) is another machine learning algorithm that is used to classify data.
+The point of SVM's are to try and find a line or **hyperplane** to divide a dimensional space which best classifies
+the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: would try `to` best separate

File: docs/source/content/overview/linear_SVM.rst [hunk 10]
```diff
@@ -2,67 +2,157 @@
...
+A **Support Vector Machine** (SVM for short) is another machine learning algorithm that is used to classify data.
+The point of SVM's are to try and find a line or **hyperplane** to divide a dimensional space which best classifies
+the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a 
+line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: ~So~
`other side`

File: docs/source/content/overview/linear_SVM.rst [hunk 11]
```diff
@@ -2,67 +2,157 @@
...
+A **Support Vector Machine** (SVM for short) is another machine learning algorithm that is used to classify data.
+The point of SVM's are to try and find a line or **hyperplane** to divide a dimensional space which best classifies
+the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a 
+line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B. 
+This alogorithm is very useful in classifying because we just have to caluclate the best line or hyperplane once
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: `calculate`

File: docs/source/content/overview/linear_SVM.rst [hunk 12]
```diff
@@ -2,67 +2,157 @@
...
+A **Support Vector Machine** (SVM for short) is another machine learning algorithm that is used to classify data.
+The point of SVM's are to try and find a line or **hyperplane** to divide a dimensional space which best classifies
+the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a 
+line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B. 
+This alogorithm is very useful in classifying because we just have to caluclate the best line or hyperplane once 
+and any new data points we can easily classify just by seeing which side of the line it falls on. Unlike KNN, where
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: points ~we~ can easily `be classified` just by seeing which side of the line `they fall` on

File: docs/source/content/overview/linear_SVM.rst [hunk 13]
```diff
@@ -2,67 +2,157 @@
...
+A **Support Vector Machine** (SVM for short) is another machine learning algorithm that is used to classify data.
+The point of SVM's are to try and find a line or **hyperplane** to divide a dimensional space which best classifies
+the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a 
+line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B. 
+This alogorithm is very useful in classifying because we just have to caluclate the best line or hyperplane once 
+and any new data points we can easily classify just by seeing which side of the line it falls on. Unlike KNN, where 
+we would have to calculate each data points nearest neighbors.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst:11]: This last sentence is missing an independent clause (full sentence) before `, where`. IE, replacing the `, where` with a period should produce two full sentences. I'd reword the first part to something like "This contrasts with the k-nearest neighbors algorithm, where ..."
each data point`'`s

File: docs/source/content/overview/linear_SVM.rst [hunk 14]
```diff
@@ -2,67 +2,157 @@
...
+The point of SVM's are to try and find a line or **hyperplane** to divide a dimensional space which best classifies
+the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a 
+line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B. 
+This alogorithm is very useful in classifying because we just have to caluclate the best line or hyperplane once 
+and any new data points we can easily classify just by seeing which side of the line it falls on. Unlike KNN, where 
+we would have to calculate each data points nearest neighbors. 
+
+Hyperplane
+----------
+A **hyperplane** depends on the space it is in, but it divides the space into two disconnected parts. So
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: Replace `So` with `For example, `

File: docs/source/content/overview/linear_SVM.rst [hunk 15]
```diff
@@ -2,67 +2,157 @@
...
+the data points. If we were trying to divide two classes A and B we would try best separate the two classes with a 
+line. So on one side of the line/hyperplane would be data from class A and on the otherside would be from class B. 
+This alogorithm is very useful in classifying because we just have to caluclate the best line or hyperplane once 
+and any new data points we can easily classify just by seeing which side of the line it falls on. Unlike KNN, where 
+we would have to calculate each data points nearest neighbors. 
+
+Hyperplane
+----------
+A **hyperplane** depends on the space it is in, but it divides the space into two disconnected parts. So 
+1-dimensional space this would just be a point, 2-d space a line, 3-d space a plane, and so on.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: ~this~

File: docs/source/content/overview/linear_SVM.rst [hunk 16]
```diff
@@ -2,67 +2,157 @@
...
+Hyperplane
+----------
+A **hyperplane** depends on the space it is in, but it divides the space into two disconnected parts. So 
+1-dimensional space this would just be a point, 2-d space a line, 3-d space a plane, and so on. 
 
 How do we find the best hyperplane/line?
 ----------------------------------------
 
-You might be wondering that there could be multiple lines that split the data well. As you can see in the graph below every line splits the squares and the circles, so which one do we choose?
+You might be wondering that there could be multiple lines that split the data well. In fact there is a infinite
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: wondering `if` there
`Surprisingly,` there is `an` infinite

File: docs/source/content/overview/linear_SVM.rst [hunk 17]
```diff
@@ -2,67 +2,157 @@
...
+amount of lines that can divide two classes.  As you can see in the graph below every line splits the squares and
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: Should be one space between sentences,
As you can see in the graph below`,` every

File: docs/source/content/overview/linear_SVM.rst [hunk 18]
```diff
@@ -2,67 +2,157 @@
...
-.. figure:: https://github.com/machinelearningmindset/machine-learning-for-everybody/blob/master/docs/source/content/overview/_img/Possible_hyperplane.PNG
-   :scale: 50 %
+.. figure:: _img/Possible_hyperplane.png
+   :scale: 50%
    :alt: Possible_Hyperplane
 
    Ref: https://towardsdatascience.com/support-vector-machine-introduction-to-machine-learning-algorithms-934a444fca47 
 
-We want to choose the line/plane with the **maximum margin**. Maximizing the margin will give us the optimal line to classify the data. This is shown in the figure below.  
+So how does SVM find the ideal line to separate the two classes. It doesn't just pick a random one. The algorithm
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: First sentence is a question (?)
The algorithm `chooses`

File: docs/source/content/overview/linear_SVM.rst [hunk 19]
```diff
@@ -2,67 +2,157 @@
...
+   :scale: 50%
    :alt: Optimal_Hyperplane
 
    Ref: https://towardsdatascience.com/support-vector-machine-introduction-to-machine-learning-algorithms-934a444fca47 
 
 How to maximize the margin?
 ---------------------------
 
-The data that is closest to the line are what determmine the optimal line. These data points are called **support vectors**. They are shown as the filled in squares and circles above. There is a lot of complex math that goes into finding the support vectors and maximizing the margin. We won't go into that, we just want to get the basic idea behind SVMs. 
+The data that is closest to the line are what determmine the optimal line. These data points are called
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: `is` what determine`s`

File: docs/source/content/overview/linear_SVM.rst [hunk 20]
```diff
@@ -2,67 +2,157 @@
...
+**support vectors**. They are shown as the filled in squares and circles above. The distance from these vectors to the
+hyperplane is called the **margin**. In general the further those points are from the hyperplane, the greater the 
+probability of correctly classifying the data. There is a lot of complex math that goes into finding the support vectors
+and maximizing the margin. We won't go into that, we just want to get the basic idea behind SVMs. 
 
 Ignore Outliers
 ---------------
 
-Sometimes data classes will have **outliers**, (data that is separated from it's class). Support Vector Machines will ignore these outliers. This is shown in the figure below. 
+Sometimes data classes will have **outliers**. This is data points that are clearly separated from the rest of it's class.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: `These are` data points that are clearly separated from the rest of `their` class

File: docs/source/content/overview/linear_SVM.rst [hunk 21]
```diff
@@ -2,67 +2,157 @@
...
-   :scale: 50 %
+
+.. figure:: _img/SVM_Outliers.png
+   :scale: 50%
    :alt: Outliers
 
    Ref:  https://www.analyticsvidhya.com/blog/2017/09/understaing-support-vector-machine-example-code/
 
-The star that is with the red circles is the outlier. So the SVM ignores the outlier and creates the best line to separate the two classes. 
+The star that is with the red circles is the outlier. So the SVM ignores the outlier and creates the best line to separate
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: ~So~

File: docs/source/content/overview/linear_SVM.rst [hunk 22]
```diff
@@ -2,67 +2,157 @@
...
+   :scale: 50%
    :alt: Optimal_Hyperplane
 
    Ref: https://towardsdatascience.com/support-vector-machine-introduction-to-machine-learning-algorithms-934a444fca47 
 
 How to maximize the margin?
 ---------------------------
 
-The data that is closest to the line are what determmine the optimal line. These data points are called **support vectors**. They are shown as the filled in squares and circles above. There is a lot of complex math that goes into finding the support vectors and maximizing the margin. We won't go into that, we just want to get the basic idea behind SVMs. 
+The data that is closest to the line are what determine the optimal line. These data points are called
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: line `is` what `determines`

File: docs/source/content/overview/linear_SVM.rst [hunk 23]
```diff
@@ -2,67 +2,157 @@
...
+**support vectors**. They are shown as the filled in squares and circles above. The distance from these vectors to the
+hyperplane is called the **margin**. In general, the further those points are from the hyperplane, the greater the 
+probability of correctly classifying the data. There is a lot of complex math that goes into finding the support vectors
+and maximizing the margin. We won't go into that; we just want to get the basic idea behind SVMs. 
 
 Ignore Outliers
 ---------------
 
-Sometimes data classes will have **outliers**, (data that is separated from it's class). Support Vector Machines will ignore these outliers. This is shown in the figure below. 
+Sometimes data classes will have **outliers**. This is data points that are clearly separated from the rest of its class.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: `These are` data points that are clearly separated from the rest of `their` class.

File: docs/source/content/overview/linear_SVM.rst [hunk 24]
```diff
@@ -2,67 +2,157 @@
...
-   :scale: 50 %
+.. figure:: _img/SVM_Kernal.png
+   :scale: 50%
    :alt: Outliers
 
    Ref:  https://www.analyticsvidhya.com/blog/2017/09/understaing-support-vector-machine-example-code/
 
 
-There's is no clear way to separate the stars from the circles. We will go over how SVMs will be able to classify non-lenearly separable data in our section about **Kernal SVM**. 
+There's is no clear way to separate the stars from the circles. SVMs will be able to classify non-linearly separable
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: `There is`

File: docs/source/content/overview/linear_SVM.rst [hunk 25]
```diff
@@ -2,67 +2,157 @@
...
+data by using a trick called the **kernel trick**. Basically, what goes behind the scenes, is that it takes the points
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: Basically, the kernel trick takes the points to a higher dimension to turn non-linearly ...

File: docs/source/content/overview/linear_SVM.rst [hunk 26]
```diff
@@ -2,67 +2,157 @@
...
+to a higher dimension, which turns non-linearly separable date to linear separable data. So the above figure would be
+classified with a circle that separates the data.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst:83]: We should include the new image from the kernel trick from that same ref: https://www.analyticsvidhya.com/wp-content/uploads/2015/10/SVM_10.png

File: docs/source/content/overview/linear_SVM.rst [hunk 27]
```diff
@@ -2,67 +2,157 @@
...
+to a higher dimension, which turns non-linearly separable date to linear separable data. So the above figure would be
+classified with a circle that separates the data. 
+
+There are three types of kernels:
+
+- **Linear** Kernel
+- **Polynomial** Kernel
+- **Radial Basis Function (RBF)** kernel
+
+You can see how these kernels change the outcome of the optimal hyperplane by changing the value of kernel in
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst:100]: This last part should go somewhere in the section where we talk about the code

File: docs/source/content/overview/linear_SVM.rst [hunk 28]
```diff
@@ -2,67 +2,157 @@
...
+
+- Performs poorly when there is a large data set, the training times are longer.
+- Performs badly when the classes are overlapping, i.e. non-separable data points.   
+
+
+Motivation
+----------
+
+Why would you ever use SVMs? There are so many different models that can classify data. Why use this one? 
+This is probably the best classifier if you know the data points are easily separable. Also it can be extended
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: Also`,`

File: docs/source/content/overview/linear_SVM.rst [hunk 29]
```diff
@@ -2,67 +2,157 @@
...
+by using kernel tricks, so try using the different kernels like Radial Basis Function (RBF). 
+
+
+Code Example
+-------------
+Check out our code, `linear_svm.py`_ to learn how to implement a linear SVM using Python's Scikit-learn library. 
+More information about `Scikit-Learn`_ can be found here.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: `here` should be the link in this scenario

File: docs/source/content/overview/linear_SVM.rst [hunk 30]
```diff
@@ -2,67 +2,157 @@
...
+It also circles the support vectors that determine the hyperplane. The output should look like this:
+
+.. figure:: _img/linear_svm_output.png
+   :scale: 50%
+   :alt: Linear SVM output
+
+The green points are classified as benign.
+The red points are classified as malignant.
+
+This loads the data from the Scikit-Learn's dataset library. You can change the data to whatever you would like
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: you would like`.`

File: docs/source/content/overview/linear_SVM.rst [hunk 31]
```diff
@@ -2,67 +2,157 @@
...
+just make sure you have, data points and an array of targets to classify those data points.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: `Just` make sure you have data points and

**Review by mergu — CHANGES_REQUESTED** [review]
These are some grammatical errors I noticed, I think I got them all. Also wanted to include the kernel trick image

File: docs/source/content/overview/linear_SVM.rst [hunk 32]
```diff
@@ -114,13 +130,13 @@ Motivation
...
 Why would you ever use SVMs? There are so many different models that can classify data. Why use this one? 
-This is probably the best classifier if you know the data points are easily separable. Also it can be extended
+This is probably the best classifier if you know the data points are easily separable. Also, it can be extended
 by using kernel tricks, so try using the different kernels like Radial Basis Function (RBF). 
 
 
 Code Example
 -------------
-Check out our code, `linear_svm.py`_ to learn how to implement a linear SVM using Python's Scikit-learn library. 
+Check out our code, `here`_ to learn how to implement a linear SVM using Python's Scikit-learn library.
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: Revert this back to linear_svm.py with the link. In my comment I meant the scikit learn link. The `here` should be the link and not the Scikit-Learn

File: docs/source/content/overview/linear_SVM.rst [hunk 33]
```diff
@@ -72,16 +72,32 @@ separable data**. Here is an example of that kind of data.
...
-data by using a trick called the **kernel trick**. Basically, what goes behind the scenes, is that it takes the points
-to a higher dimension, which turns non-linearly separable date to linear separable data. So the above figure would be
+There is no clear way to separate the stars from the circles. SVMs will be able to classify non-linearly separable
+data by using a trick called the **kernel trick**. Basically, the kernel trick takes the points
+to a higher dimension to turn non-linearly separable data to linear separable data. So the above figure would be
 classified with a circle that separates the data. 
 
+Here is an example of the kernel trick.
+
+.. figure:: _img/SVM_Kernel1.png
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst]: This image can be removed, just use the circled one

File: docs/source/content/overview/linear_SVM.rst [hunk 34]
```diff
@@ -72,16 +72,32 @@ separable data**. Here is an example of that kind of data.
...
+   :scale: 50%
+   :alt: Kernel X Z graph
+
+   Ref:  https://www.analyticsvidhya.com/blog/2017/09/understaing-support-vector-machine-example-code/
+
+.. figure:: _img/SVM_Kernel2.png
```
Comment (mergu) [inline docs/source/content/overview/linear_SVM.rst:87]: This image doesn't seem to be scaled the same as the others. Can you copy it directly from the ref and make sure it's the same size?

**Review by mergu — APPROVED** [review]

**Review by astorfi — CHANGES_REQUESTED** [review]
This branch has to merge conflicts ... Please investigate those ... Alternatively, you can remove this pull request and pull request again with your latest changes.

**Comment by mergu** [comment]
This is being closed as a PR without merge conflicts was created.
